src/browser/page.zig

```diff
@@ -467,12 +467,15 @@ pub const Page = struct {
         const owned_url = try self.arena.dupeZ(u8, request_url);
         self.url = try URL.parse(owned_url, null);
 
+        var headers = try HttpClient.Headers.init();
```

Comment:
A lot of internal failure cases will result in this leaking. But I think we can leave it as-is for now.

The difficulty in resource management here is that we need to be careful about allocations up to the point where `curl_multi_add_handle` is called. After that, it's handled by the `perform` loop. So a simple `errdefer` won't do, since you risk double-free depending on where the failure is. A simple fix is to not call` perform `in `makeRequest` - making `curl_multi_add_handle` the last-called function - but I feel like delaying `perform` until the next `tick` (especially since there's no reason to think the request can't be sent out immediately) is unnecessary latency.

I want to address it, but, at this point, after we merge everything.